### PR TITLE
[MOS-893] Fix invalid time description in the SMS thread

### PR DESCRIPTION
**Pull request by Maciej-Mudita**
The sending/receiving time information will be displayed as follows:
- Only an hour –> for today
- Yesterday and hour –> for messages received/sent yesterday
- Day of the week and hour –> for messages received/sent within last 7 days
- Date (day+month) and hour –> for messages older than week, but still from the same year
- Date (day+month+year) and hour –> for messages from previous years

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/application-messages/models/ThreadsSearchResultsModel.cpp [hunk 1]
```diff
@@ -35,7 +35,8 @@ namespace gui::model
         {
             using namespace utils::time;
             ret->setContact(threadStruct->contact->getFormattedName());
-            ret->setTimestamp(*TimestampFactory().createTimestamp(TimestampType::DateTime, threadStruct->thread->date));
+            ret->setTimestamp(
+                *TimestampFactory().createTimestamp(TimestampType::DateOrTime, threadStruct->thread->date));
```
Comment (marziel) [inline module-apps/application-messages/models/ThreadsSearchResultsModel.cpp:39]: Couldn't it be replaced with `->`?

File: module-utils/time/time/time_conversion.cpp [hunk 2]
```diff
@@ -207,15 +207,44 @@ namespace utils::time
...
-        auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM_YYYY
-                                                            : Locale::TimeFormat::FormatLocaleDate_MM_DD_YYYY;
+        const auto localeFormat = timeSettings.isDateFormatDDMM() ? Locale::TimeFormat::FormatLocaleDate_DD_MM_YYYY
+                                                                  : Locale::TimeFormat::FormatLocaleDate_MM_DD_YYYY;
         return Timestamp::str(Locale::format(localeFormat));
     }
 
+    UTF8 DateAndTime::str(std::string format) const
+    {
+        if (format.compare("") != 0) {
```
Comment (marziel) [inline module-utils/time/time/time_conversion.cpp:221]: Maybe better just `empty()`? Or is there a special purpose to do it this way here?

**Review by marziel — APPROVED** [review]

File: module-apps/application-messages/models/ThreadsSearchResultsModel.cpp (hunk 1 above)
Comment (Maciej-Mudita) [reply module-apps/application-messages/models/ThreadsSearchResultsModel.cpp]: done

File: module-utils/time/time/time_conversion.cpp (hunk 2 above)
Comment (Maciej-Mudita) [reply module-utils/time/time/time_conversion.cpp]: done

**Review by Lefucjusz — APPROVED** [review]

**Pull request merged**